content/working_with/official_plugins/Orchestration/terraform.md

```diff
@@ -86,19 +86,35 @@ This refers to a Terraform Plan module.
-  * `terraform.refresh`: Refresh Terraform state file, if any changes were done outside of terraform so it will update the runtimes properties to match the real properties for the created resources
+    * `source`: the new template location by default the `last_source_location` but it can be changed to be another location or even URL to a new template
+    * `destroy_previous`: boolean if set to True it will trigger destroy for the previously created resources , if False it will keep them and maintain the state file , and terraform will calculate the changes needed to be applied to those already created resources
+  * `terraform.refresh`: Refresh Terraform state file, if any changes were done outside of terraform so it will update the runtime properties to match the real properties for the created resources under `state` runtime property. 
+    Moreover, If there are any drifts between the template and the current state it will be saved under `drifts` runtime property.
 
+**Runtime Properties**:
+
+ * `state`: Saves the state of the resources created in the format { "resource_name" : <resource state> }, 
+   <resource state> is the state of the resource that pulled with `terraform state pull` command.
+   
+ * `drifts`: Saves the drifts between the template and the current state in the format:
+    { "resource_name" : <change-representation> }, <change-representation> format described [here](https://www.terraform.io/docs/internals/json-format.html#change-representation).
+ 
+ * `is_drifted`: True if there are drifts between the template and the actual state, else False.
+
+ * `terraform_source`: Base64 encoded representation of the zip contains the terrafom modules. 
 
 **Workflows**
 
-  * `refresh_terraform_resources`: execute `terraform.refresh` operation on `terraform.Module` node instances
-  * `reload_terraform_template`: executes `terraform.reload` on `terraform.Module` node instances
+  * `refresh_terraform_resources`: execute `terraform.refresh` operation on `terraform.Module` node instances.
+  * `reload_terraform_template`: executes `terraform.reload` on `terraform.Module` node instances.
+
+**Notes:**
 
-By default, the aforementioned workflows operate on all `terraform.Module` node instances in the current deployment.
+* By default, the aforementioned workflows operate on all `terraform.Module` node instances in the current deployment.
 It is possible to limit the scope by using the `node_ids` and `node_instance_ids` parameters, specifying lists of
 node ID's and node instance ID's to operate on.
+* `pull` workflow introduced on cloudify version 5.3. Since version 0.16.0, Terrafom plugin introduced pull operation for `terraform.Module` node in order to support pull workflow.
```

Comment:
@ofercloudify @EarthmanT  the version param points to the 5.1.1 version. I think that we introduced pull workflow in version 5.3 but I don't know if it's a good idea to write this as I wrote. Do you have any ideas?

Reply:
You could also put, "for versions before 5.1.1 call this operation with execute operation workflow.".